### PR TITLE
[CI] Automate the merge of the release branches

### PR DESCRIPTION
**Pull request by testableapple**
### 🔗 Issue Links

Resolve https://stream-io.atlassian.net/browse/PBE-5401

### 🎯 Goal

Automate the merge of the release branches

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -140,11 +147,11 @@ lane :merge_main_to_develop do
 
   sh('git checkout main')
   sh('git pull origin main')
-  sh('git checkout develop')
+  sh('git checkout origin/develop')
   sh('git pull origin develop')
   sh('git log develop..main')
   sh('git merge main')
-  sh('git push')
+  sh('git push origin HEAD:develop')
```
Comment (testableapple) [inline fastlane/Fastfile:154]: Tested this locally on SwiftUI and it worked fine. Still needs to be tested on CI on release.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -116,17 +116,24 @@ end
...
+
+  sh('git checkout origin/main')
+  sh('git pull origin main')
 
   # Merge release branch to main. For more info, read: https://notion.so/iOS-Branching-Strategy-37c10127dc26493e937769d44b1d6d9a
-  sh("git merge #{release_branches.first} --ff-only")
-  UI.user_error!('Not pushing changes') unless prompt(text: 'Will push changes. All looking good?', boolean: true)
-  sh('git push')
+  sh("git merge #{release_branch} --ff-only")
+  sh('git push origin HEAD:main')
```
Comment (testableapple) [inline fastlane/Fastfile:136]: I assume it will work similar to `merge main to develop`, so also should be tested on release.

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**